### PR TITLE
Make unreachable else in case statements a warning instead of an error

### LINKED ISSUE #14019: Add pragma for disabling enforced exhaustive case coverage
### Problem

Enforcing exhaustive coverage in case statement is an awesome feature, but sometimes you don't know if you have covered all cases or not. This creates the following situation where there is always a possibility to get an error:

- If you **don't** add an `else: discard` statement -> `not all cases are covered
`
- If you **do** add an `else: discard` statement ----> `invalid else, all cases are already covered`

This of course is no problem during the general development case, because the compiler will just inform you whether you are covering all cases or not, which is great. In case you are generating code though, getting such compile errors when you know your generated code is correct can be bothersome.

### Possible solutions

1. Introduce a pragma for disabling the check of exhaustive coverage. Possible names:
  - `{.dirtyCase.}`
  - `{.lenientCase.}`

2. Change `Error: invalid else, all cases are already covered` into a warning.

### PR DESCRIPTION
**Pull request by Clyybber**
Fixes #14019

### CODE REVIEW

File: compiler/lineinfos.nim [hunk 1]
```diff
@@ -96,6 +96,7 @@ const
     warnProveInit: "Cannot prove that '$1' is initialized. This will become a compile time error in the future.",
     warnProveField: "cannot prove that field '$1' is accessible",
     warnProveIndex: "cannot prove index '$1' is valid",
+    warnTrailingElse: "trailing else, all cases are already covered",
```
Comment (disruptek) [inline compiler/lineinfos.nim]: I think I'd speak more to the problem you're warning about; "else clause ignored" or some such.

**Review by disruptek — APPROVED** [review]

File: compiler/lineinfos.nim (hunk 1 above)
Comment (Clyybber) [reply compiler/lineinfos.nim]: Renamed it to UnreachableElse now.

**Comment by kaushalmodi** [comment]
@Clyybber Can you also rename that in the commit log and the PR title?

**Comment by timotheecour** [comment]
@Clyybber IMO this would be much better if nim defaults to `warningAsError:warnUnreachableElse:on` and then the occasional use case can simply use
```nim
{.push warningAsError:warnUnreachableElse:off.}
code
{.pop}
```
there's really no need for this not to be an error outside of the code generation case

**Comment by Clyybber** [comment]
@timotheecour Theres no need for it to be an error in any case, if you want it to be an error use warningAsError, but I think defaulting some warnings to be errors is a really bad idea.
To give an example where you really wouldn't want it to be an error outside of code generation:
Multiple people are working on something, lets say a VM. The opcodes are enums and there is an `echo "NYI"` in the else branch.
Now when someone comes along and implements all codes that else will be an error which is just annoying, since it could very well be that the next commit will introduce a new opcode.

Changing it to be an error would gain us nothing and would just make the default more annoying.
And worst of all it would introduce true language dialects, since then a library might only compile with warningsAsErrors:warnUnreachableElse:off

**Comment by timotheecour** [comment]
ok, fair enough!

**Pull request merged**